eclipse-glsp/glsp-theia-integration
# GLSP-1357: Update GLSPSaveable for Theia 1.50.x

#### What it does


With Theia 1.50 the `Saveable` implementation was changed in breaking fashion. In particular the `Saveable` itself is no longer responsible of handling autosave behavior. This change updates the `GLSPSaveable` implementation to be compatible with Theia > 1.50.x while maintaining the old saveable behavior for Theia < 1.50

- Introduce utility functions to determine the currently used Theia version and validate whether the version satisfies a given range
- Update `GLSPSaveable` implementation to support the `onContentChanged` event required for Theia 1.50.x.  The old autosave behavior is kept and only activated if a Theia version < 1.50.0 is used.
- Update READMEs

- Introduce a utility script to change the Theia version used in the example applications
- Add CI workflows that run periodicaly (once a week)
  - `Multi OS/Node` Builds against all OS variants in combination with node 18 and 20
  - `Theia Compatibility` Builds the current master/examples again the minimum/CR/latest Theia version

Fixes https://github.com/eclipse-glsp/glsp/issues/1357
#### How to test
Build and start the example. Verify that autosaving in the diagram still works as expected.
Switch the example applications to Theia 1.50.1 (`yarn change:theia-version 1.50.1`)
Rebuild and retest


#### Follow-ups



#### Changelog
